test_scripts/MobileProjection/Phase1/common.lua

```diff
@@ -78,6 +80,83 @@ function m.StopStreaming(pService, pFile, pAppId)
   end
 end
 
+--[[ @ValidateStreamingData: Validate HMI bound streaming data
+--! @parameters:
+--! event - Streaming data captured event
+--! compare_file - file that was streamed to compare data against what was received
+--! @return: none
+--]]
+function m.ValidateStreamingData(event, compare_file)
+  EXPECT_EVENT(event, "Stream event")
+  :Do(function(_, data)
+    if data.success then
+      local recv_file = io.open(data.file_name, "rb")
+      local recv_data = recv_file:read(data.total_bytes)
+      recv_file:close()
+
+      local sent_file = io.open(compare_file, "rb")
+      local sent_data = sent_file:read(data.total_bytes)
+      sent_file:close()
+
+      if sent_data ~= recv_data then
+        actions.run.fail("Streaming Data Received by HMI did not match the file sent")
+      end
+    end
+  end)
+end
+
+--[[ @ListenStreaming: Capture HMI bound streaming data to file
+--! @parameters:
+--! pService - service value
+--! bytes - how many bytes before the callback will be called
+--! compare_file - file to compare received data to
+--! @return: event which will be raised when stream data is received or times out
+--]]
+function m.ListenStreaming(pService, bytes, compare_file)
+  local consumer = "socket"
+  if pService == 10 then
+    consumer = actions.sdl.getSDLIniParameter("AudioStreamConsumer")
+  elseif pService == 11 then
+    consumer = actions.sdl.getSDLIniParameter("VideoStreamConsumer")
+  end
+  local isTcp = "socket" == consumer
+
+  local event = events.Event()
+  event.service = pService
+  event.matches = function(event1, event2)
+    return event1.service == event2.service
+  end
+
+  if compare_file then
+    m.ValidateStreamingData(event, compare_file)
+  end
+
+  local callback = function(success, total_bytes, file_name)
+    event.success = success
+    event.total_bytes = total_bytes
```

Comment:
I am printing out the total_bytes and am getting a variable amount. Most of the times it prints 65536. Some times it prints 32768.

Reply:
Those are TCP frame sizes and due to the way the socket is being read. I have updated the amount of bytes that are being validated in https://github.com/smartdevicelink/sdl_atf_test_scripts/pull/2568/commits/b63464e7b2fa1f477aef7f485e03a042df7ac2dc so that multiple frames will always be evaluated.